examples/chip-tool/commands/interactive/InteractiveCommands.cpp

```diff
@@ -153,47 +153,59 @@ struct InteractiveServerResult
     {
         auto lock = ScopedLock(mMutex);
 
-        std::string resultsStr;
+        std::stringstream results;
         if (mResults.size())
         {
             for (const auto & result : mResults)
             {
-                resultsStr = resultsStr + result + ",";
+                results << result << ",";
             }
 
             // Remove last comma.
-            resultsStr.pop_back();
+            results.seekp(-1, std::ios_base::end);
+            results << '\0';
         }
 
         if (mStatus != EXIT_SUCCESS)
         {
-            if (resultsStr.size())
+            if (results.tellp())
             {
-                resultsStr = resultsStr + ",";
+                results << ",";
             }
-            resultsStr = resultsStr + "{ \"error\": \"FAILURE\" }";
+            results << "{ \"error\": \"FAILURE\" }";
         }
 
-        std::string logsStr;
+        std::stringstream logs;
         if (mLogs.size())
         {
             for (const auto & log : mLogs)
             {
-                logsStr = logsStr + "{";
-                logsStr = logsStr + "  \"module\": \"" + log.module + "\",";
-                logsStr = logsStr + "  \"category\": \"" + log.messageType + "\",";
-                logsStr = logsStr + "  \"message\": \"" + log.message + "\"";
-                logsStr = logsStr + "},";
+                logs << "{"
+                        "  \"module\": \"" +
+                        log.module +
+                        "\","
+                        "  \"category\": \"" +
+                        log.messageType +
+                        "\","
+                        "  \"message\": \"" +
+                        log.message +
+                        "\""
+                        "},";
             }
 
             // Remove last comma.
-            logsStr.pop_back();
+            logs.seekp(-1, std::ios_base::end);
+            logs << '\0';
         }
 
+        auto resultsString  = results.str();
+        auto logsString     = logs.str();
+        auto resultsContent = resultsString.c_str();
+        auto logsContent    = logsString.c_str();
         std::string jsonLog;
         jsonLog = jsonLog + "{";
```

Comment:
you are doing concatenations here too with large strings. the first `+` is with an empty string so is not needed, the rest seem like they would also benefit from stringstream to not reallocate as often (although this reallocates just 2 times).

Also if we just need a single stream, why don't we only have a single stringstream and do all the writing in it without the intermediate logs and results strings?

Reply:
I have updated the code such that it does use a single string stream. Thanks !